extensions/wikia/Search/classes/Services/QuestDetails/QuestDetailsSolrHelper.class.php

```diff
@@ -83,39 +84,46 @@ public function consumeResponse( $response ) {
 
 	public function addCategories( $resultArray, $categoriesArray ) {
 		foreach ( $resultArray as $key => $item ) {
-			if ( isset( $categoriesArray[ $item[ 'id' ] ] ) ) {
-				$resultArray[ $key ][ 'categories' ] = $categoriesArray[ $item[ 'id' ] ];
+			if ( isset( $categoriesArray[$item['id']] ) ) {
+				$resultArray[$key]['categories'] = $categoriesArray[$item['id']];
 			}
 		}
+
 		return $resultArray;
 	}
 
 	public function filterIdsByCategory( array $categriesIDArray, $category ) {
 		foreach ( $categriesIDArray as $id => $categories ) {
 			if ( !in_array( $category, $categories ) ) {
-				unset ( $categriesIDArray[ $id ] );
+				unset ( $categriesIDArray[$id] );
 			}
 		}
+
 		return $categriesIDArray;
 	}
 
 	public function findCategoriesForIds( array $ids ) {
 		$db = wfGetDB( DB_SLAVE );
-		$out = ( new WikiaSQL() )
-			->SELECT( "cl_to, cl_from" )
+		$out = ( new WikiaSQL() )->SELECT( "cl_to, cl_from" )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$out". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)